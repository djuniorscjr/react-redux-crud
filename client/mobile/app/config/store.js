import { createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import reduxThunk from 'redux-thunk';

import reducers from '../reducers';

const middleware = [reduxThunk];

if (process.env.NODE_ENV === 'development') {
	middleware.push(logger);
}

export default createStore(reducers, applyMiddleware(...middleware));
