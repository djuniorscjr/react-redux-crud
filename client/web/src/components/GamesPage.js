import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import GamesList from './GamesList';
import { fetchGames, deleteGame } from '../actions'

class GamesPage extends React.Component {
	componentDidMount() {
		this.props.fetchGames();
	}

	render(){
		return (
			<div>
				<h2>Games List</h2>
				<GamesList games={this.props.games} deleteGame={this.props.deleteGame} />
			</div>
		);
	}
}

GamesPage.propTypes = {
	games: PropTypes.array.isRequired,
	fetchGames: PropTypes.func.isRequired,
	deleteGame: PropTypes.func.isRequired
};

const mapStateToProps = (state) => ({
	games: state.games,
})

export default connect(mapStateToProps, { fetchGames, deleteGame })(GamesPage);
