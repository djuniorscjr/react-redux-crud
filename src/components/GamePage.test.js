import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import GamesPage from './GamesPage';
import store from '../config/store';
import { Provider } from 'react-redux';

test('GamesPage component should render as expected', () => {
	const component = shallow(
		<Provider store={store}>
			<GamesPage />
		</Provider>
	);
	const tree = toJson(component);
	expect(tree).toMatchSnapshot();
});
