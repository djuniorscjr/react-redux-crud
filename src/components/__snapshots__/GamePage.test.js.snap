// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GamesPage component should render as expected 1`] = `<Connect(GamesPage) />`;
